### Домашнее задание №2
***Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона***

In [1]:
from math import factorial
import numpy as np

def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

***1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.***

*Вероятность находится по формуле Бернулли:* $P_n^k= C_n^kp^kq^{n-k}$
*где* $q=1-p$.

*В нашем случае $n = 100$, $k=85$, $p=0,8$, $q=0,2$. *

Получается $P_{100}^{85}= C_{100}^{85}\cdot0,8^{85}\cdot0,2^{100-85}$$

In [2]:
P=combinations(100, 85)*(0.8**85)*(0.2**15)
P

0.048061793700746556

***Ответ:*** *0.048061793700746556*

***2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?***

*Так как вероятность события постоянна и мала, а количество испытаний велико, вместо ф.Бернули проще применинить формулу Пуассона:*

## $$P_m \approx \frac{\lambda^m}{m!}e^{-\lambda}$$

*где среднее число появлений событий* $\lambda=np$.

*Получаем при $n=5000$, $p=0,0004$, $\lambda=np=5000\cdot{0,0004}=2$.*

*Вероятность, что ни одна из лампочек не перегорит в первый день: $P_0 \approx \frac{2^0}{0!}e^{-2}$*

In [3]:
p1=(2**0/factorial(0))*np.exp(-2)
p1

0.1353352832366127

*Вероятность, что перегорят две лампочки: $P_2 \approx \frac{2^2}{2!}e^{-2}$*

In [4]:
p2=(2**2/factorial(2))*np.exp(-2)
p2

0.2706705664732254

***Ответ:*** *а) $P_1 = 0.1353352832366127$;* *б) $P_2 = 0.2706705664732254$*

***3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?***

*вероятность находится по формуле Бернулли:*

## $$P_n^k= C_n^kp^kq^{n-k}$$

*где $q=1-p$.*

*В нашем случае получается $n = 144$, $k=70$, $p=0,5$, $q=0,5$. что верятность*: $P_{144}^{70}= C_{144}^{70}\cdot0,5^{70}\cdot0,5^{144-70}$ 

In [5]:
P=combinations(144, 70)*(0.5**70)*(0.5**74)
P

0.06281178035144776

***Ответ:*** *0.06281178035144776*

***4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?***

a) Вероятность того, что из первого ящика будет вытащено 2 белых мяча:

In [6]:
(7/10)*(6/9)

0.4666666666666666

Вероятность того, что из второго ящика будет вытащено 2 белых мяча:

In [7]:
(9/11)*(8/10)

0.6545454545454547

Вероятность того, что все мячи белые:

In [8]:
(7/10)*(6/9)*(9/11)*(8/10)

0.3054545454545454

*b) Опишем все возможные благоприятные события*:

<li><i>Два белых мяча из первого ящика и два черных мяча из второго ящика;
<li><i>По одному белому мячу из каждого ящика и по одному черному мячу из каждого ящика;
<li><i>Два черных мяча из первого ящика и два белых мяча из второго ящика;

Вероятность того, что из второго ящика вытащено 2 чёрных мяча:

In [9]:
(2/11)*(1/10)

0.018181818181818184

Общая вероятность того, что 2 мяча из первого ящика белые и 2 мяча из второго - чёрные:

In [10]:
(7/10)*(6/9)*(2/11)*(1/10)

0.008484848484848484

*Вероятность того, что один мяч из первого ящика белый и один мяч из первого ящика чёрный:*

*Возможны два благоприятных исхода при неважном для нас порядке:*
<li><i>Cначала белый потом чёрный;
<li><i>Сначала чёрный потом белый.


In [11]:
(7/10)*(3/9) + (3/10)*(7/9)

0.4666666666666667

*Вероятность того, что 1 мяч из второго ящика белый и 1 мяч из второго ящика чёрный:*

*Возможны два благоприятных исхода при неважном для нас порядке:*
<li><i>Cначала белый потом чёрный;
<li><i>Сначала чёрный потом белый.

In [12]:
(9/11)*(2/10) + (2/11)*(9/10)

0.32727272727272727

*Вероятность того, что 1 мяч из первого ящика белый и 1 мяч из первого ящика чёрный и 1 мяч из второго ящика белый и 1 мяч из второго ящика чёрный:*

In [13]:
((7/10)*(3/9) + (3/10)*(7/9))*((9/11)*(2/10) + (2/11)*(9/10))

0.15272727272727274

*Вероятность того, что из первого ящика будет вытащено 2 чёрных мяча:*

In [14]:
(3/10)*(2/9)

0.06666666666666667

Общая вероятность того, что 2 мяча из первого ящика чёрные и 2 мяча из второго - белые:

In [15]:
(3/10)*(2/9)*(9/11)*(8/10)

0.04363636363636364

Вероятность того, что два мяча белые:

In [16]:
(7/10)*(6/9)*(2/11)*(1/10) + ((7/10)*(3/9) + (3/10)*(7/9))*((9/11)*(2/10) + (2/11)*(9/10)) + (3/10)*(2/9)*(9/11)*(8/10)

0.20484848484848486

*в) Найдём вероятность того, что все мячи чёрные:*

In [17]:
(3/10)*(2/9)*(2/11)*(1/10)

0.0012121212121212121

*Так как остальные события нам подходят, то искомая вероятность будет противоположна и равна:*

In [18]:
1 - (3/10)*(2/9)*(2/11)*(1/10)

0.9987878787878788

***Ответ:*** *а) 0.3054545454545454; б) 0.20484848484848486; в) 0.9987878787878788*